demisto_sdk/dev_tools/initiator.py

```diff
@@ -0,0 +1,286 @@
+import os
+import shutil
+import yaml
+import yamlordereddictloader
+
+from typing import Dict
+from distutils.dir_util import copy_tree
+from demisto_sdk.common.tools import print_error, print_color, LOG_COLORS
+
+INTEGRATIONS_DIR = '/Integrations'
+SCRIPT_DIR = '/Scripts'
+INCIDENT_FIELDS_DIR = '/IncidentFields'
+INCIDENT_TYPES_DIR = '/IncidentTypes'
+INDICATOR_FIELDS_DIR = '/IndicatorFields'
+PLAYBOOKS_DIR = '/Playbooks'
+LAYOUTS_DIR = '/Layouts'
+TEST_PLAYBOOKS_DIR = '/TestPlaybooks'
+CLASSIFIERS_DIR = '/Classifiers'
+CONNECTIONS_DIR = '/Connections'
+DASHBOARDS_DIR = '/Dashboards'
+MISC_DIR = '/Misc'
+REPORTS_DIR = '/Reports'
+WIDGETS_DIR = '/Widgets'
+
+HELLO_WORLD_INTEGRATION = 'HelloWorld'
+HELLO_WORLD_SCRIPT = 'HelloWorldScript'
+
+
+class Initiator:
+    """Initiator creates a new pack/integration/script.
+
+    Attributes:
+        output_dir (str): The directory in which init will create the new pack/integration/script
+        name (str): The name for the new pack/integration/script
+        integration (bool): Indicates whether to create an integration.
+        script (bool): Indicates whether to create a script.
+        full_path (str): The full path to the newly created pack/integration/script
+    """
+
+    def __init__(self, output_dir: str, name: str, integration: bool = False, script: bool = False):
+        if output_dir:
+            self.output_dir = output_dir
+
+        else:
+            self.output_dir = ''
+
+        self.name = name
+        self.integration = integration
+        self.script = script
+        self.full_path = ''
+
+    def init(self):
+        """Starts the init command process.
+
+        """
+        if self.integration:
+            self.check_name(created_object="integration")
+            self.integration_init()
+
+        elif self.script:
+            self.check_name(created_object="script")
+            self.script_init()
+
+        else:
+            self.check_name(created_object="pack")
+            self.pack_init()
+
+    def check_name(self, created_object: str):
+        """Makes sure a name is given for the created object
+
+        Args:
+            created_object (str): the type of the created object (integration/script/pack)
+        """
+        while self.name is None:
+            print(f"Please input the name of the initialized {created_object}:")
+            self.name = str(input())
```

Comment:
the string from the print can be passed to the input if i'm not mistaken

Reply:
done